### PR TITLE
Refocus to app-drawer itself and close with esc

### PR DESCRIPTION
**Pull request by keanulee**

### CODE REVIEW

File: app-drawer/app-drawer.html [hunk 1]
```diff
@@ -170,6 +170,10 @@
 
       is: 'app-drawer',
 
+      hostAttributes: {
+        tabindex: '0'
+      },
+
```
Comment (frankiefu) [inline app-drawer/app-drawer.html]: What about setting `tabindex="0"` on the `#contentContainer`?  If this is something we don't really want to expose to the user then it's better to not put it on the host.

Comment (keanulee) [inline app-drawer/app-drawer.html]: Sure - I don't see a problem with that.

File: app-drawer/app-drawer.html [hunk 2]
```diff
@@ -318,19 +322,19 @@
       },
 
       _focusHandler: function(event) {
-        if (!this.isLightDescendant(Polymer.dom(event).rootTarget)) {
-          this._focusInsideDrawer();
+        if (Polymer.dom(event).path.indexOf(this.$.contentContainer) === -1) {
```
Comment (frankiefu) [inline app-drawer/app-drawer.html]: :+1:

**Comment by keanulee** [comment]
@frankiefu A user can listen for transitionend on `<app-drawer>` itself and focus on that. There are a total of 3 transitionend events (app-drawer visibility, scrim opacity, and contentContainer transform), but that can be easily read by the user.

**Comment by frankiefu** [comment]
`transitionend ` is implementation detail and I would prefer to have an event clearly indicate when the drawer is completed closed/opened.

**Comment by frankiefu** [comment]
Let's think about if we want to add a new event later.  LGTM on this PR.

**Pull request merged**